core/src/main/scala/org/apache/spark/TestUtils.scala

```diff
@@ -223,24 +224,37 @@ private[spark] object TestUtils {
 
   /**
    * Asserts that exception message contains the message. Please note this checks all
-   * exceptions in the tree.
+   * exceptions in the tree. If a type parameter `E` is supplied, this will additionally confirm
+   * that the exception is a subtype of the exception provided in the type parameter.
    */
-  def assertExceptionMsg(exception: Throwable, msg: String, ignoreCase: Boolean = false): Unit = {
-    def contain(msg1: String, msg2: String): Boolean = {
+  def assertExceptionMsg[E <: Throwable : ClassTag](
+    exception: Throwable,
```

Comment:
nit: 4 spaces indentation for method parameters.

Reply:
Argh. Try as I might, I can't convince my IDE to use 4 spaces here. Thanks for the catch.